### PR TITLE
Query log extractor for Aurora MySQL

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

Get query log from CloudWatch

### 🤓 What?

- The `logicalID` of MySQL dataset was changed incorrectly in #931, fixed in this PR.
- Refactor log extraction from CloudWatch
- MySQL crawler support query log extraction 

### 🧪 Tested?

image

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

**Review by mars-lan — COMMENTED** [review]
LGTM from a high level. @usefulalgorithm pls take a closer look.

File: metaphor/mysql/extractor.py [hunk 1]
```diff
@@ -29,3 +48,79 @@ async def extract(self) -> Collection[ENTITY_TYPES]:
...
+                query_id=sql_hash,
+                platform=DataPlatform.MYSQL,
+                default_schema=database if database else None,
+                user_id=user,
+                sql=sql,
+                sql_hash=sql_hash,
+                start_time=to_utc_datetime_from_timestamp(int(timestamp_us) / 1000_000),
+                sources=tll.sources,
+                targets=tll.targets,
+            )
```
Comment (usefulalgorithm) [inline metaphor/mysql/extractor.py:124]: I feel like this pattern shows up a lot in our code... maybe refactor this?

File: metaphor/postgresql/extractor.py [hunk 2]
```diff
@@ -403,17 +402,25 @@ async def extract(self) -> Collection[ENTITY_TYPES]:
...
-            yield from self._collect_query_logs_from_cloud_watch(
-                client=self._query_log_config.aws.get_session().client("logs"),
-                lookback_days=self._query_log_config.lookback_days,
-                logs_group=self._query_log_config.logs_group,
-            )
+            client = self._query_log_config.aws.get_session().client("logs")
+            lookback_days = self._query_log_config.lookback_days
+            logs_group = self._query_log_config.logs_group
+
+            print(iterate_logs_from_cloud_watch)
```
Comment (usefulalgorithm) [inline metaphor/postgresql/extractor.py:417]:
```suggestion
```

File: metaphor/mysql/config.py [hunk 3]
```diff
@@ -1,9 +1,36 @@
...
 from metaphor.database.config import GenericDatabaseRunConfig
 
 
+@dataclass(config=ConnectorConfig)
+class MySQLQueryLogConfig:
+    # Number of days back of query logs to fetch, if 0, don't fetch query logs
+    lookback_days: int = 1
+
+    # Query log filter to exclude certain usernames
+    excluded_usernames: Set[str] = field(default_factory=lambda: set())
```
[the remark this answers is not in the document]
Comment (elic-eon) [reply metaphor/mysql/config.py:18]: in this case, the config can overwrite the default set. I want to add `rdsadmin` to the set

File: metaphor/common/aws.py [hunk 4]
```diff
@@ -53,3 +55,42 @@ def get_session(self):
...
+        next_token = response["nextToken"] if "nextToken" in response else None
+
+        for event in response["events"]:
+            message = event["message"]
+            count += 1
+
+            if count % 1000 == 0:
+                logger.info(f"Processed {count} logs")
+
+            yield message
```
[the remark this answers is not in the document]
Comment (elic-eon) [reply metaphor/common/aws.py:93]: This is a generic function to get messages from Cloudwatch, in MySQL case, it's a CSV row but not in Postgres.

File: metaphor/mysql/extractor.py (hunk 1 above)
Comment (elic-eon) [reply metaphor/mysql/extractor.py:124]: Good point. Refactored.

**Pull request merged**